src/dmd/clone.d

```diff
@@ -404,7 +404,12 @@ bool needOpEquals(StructDeclaration sd)
 {
     //printf("StructDeclaration::needOpEquals() %s\n", sd.toChars());
     if (sd.isUnionDeclaration())
-        goto Ldontneed;
+    {
+        /* If a union has only one field, treat it like a struct
```

Comment:
How is this related to making the backend able to inline multiline functions?

Reply:
The inlining exposed a code path where a test suite double in a union got 0 initialized rather than NaN initialized.

Reply:
This kind of detail should be mentioned in the comment.

Ready to merge?

Reply:
Line 408 is the comment! Yes, pls merge.